server/users.ts

```diff
@@ -675,13 +668,8 @@ export class User extends Chat.MessageContext {
 	async rename(name: string, token: string, newlyRegistered: boolean, connection: Connection) {
 		let userid = toID(name);
 		if (userid !== this.id) {
-			for (const roomid of this.games) {
-				const room = Rooms.get(roomid);
-				if (!room || !room.game || room.game.ended) {
-					this.games.delete(roomid);
-					console.log(`desynced roomgame ${roomid} renaming ${this.id} -> ${userid}`);
-					continue;
-				}
+			for (const room of Rooms.rooms.values()) {
+				if (!room || !room.game || room.game.ended) continue;
 				if (room.game.allowRenames || !this.named) continue;
 				this.popup(`You can't change your name right now because you're in ${room.game.title}, which doesn't allow renaming.`);
```

Comment:
You forgot to test that the user is actually in the game, which will make renaming always fail in big servers like Main.

```suggestion
			for (const game of this.getGames()) {
				if (game.ended) continue;
				if (game.allowRenames || !this.named) continue;
				this.popup(`You can't change your name right now because you're in ${room.game.title}, which doesn't allow renaming.`);
```